### DISCUSSION TITLE
How Releases Work

### DISCUSSION THREAD

**Discussion by pzivich** [pzivich/zepid#154] [General]
This thread is how the releases work for zEpid. This is both for recording how it operates, and as a reminder for myself (when I inevitably take a long time for the next release)

Once the version is completed, I take the following steps:

- Test locally with `pytest tests/`
- Push to GitHub and check travis-ci tests also pass
- Generate and check local documentation via `sphinx-build -b html docs builddir`
- Generate distribution files via `python setup.py sdist bdist_wheel`
- Finally upload to PyPI via `python -m twine upload -r pypi dist/*`

**Comment by CamDavidsonPilon** [comment]
:wave: the last two steps can be automated with GH actions - I started [doing this for lifelines](https://github.com/CamDavidsonPilon/lifelines/blob/master/.github/workflows/pythonpublish.yml), and it's been much smoother. You give your twine user/pass securely to GH, too (forget how though). The workflow is triggered on [creating a new GH release](https://github.com/CamDavidsonPilon/lifelines/blob/master/.github/workflows/pythonpublish.yml#L8), which was already part of the lifelines-new-release workflow.

**Reply by pzivich** [reply]
Oh that's good! I will have to look into it. Thanks!
